kratos/utilities/read_materials_utility.cpp

```diff
@@ -296,66 +280,67 @@ void ReadMaterialsUtility::CreateProperty(
 
 void ReadMaterialsUtility::CreateSubProperties(
     ModelPart& rModelPart,
-    Parameters Data,
-    Properties::Pointer& pNewProperty
+    const Parameters SubPropertiesData,
+    Properties& rProperty
     )
 {
     KRATOS_TRY;
 
-    if (Data.Has("sub_properties")) {
-
-        auto& r_list_sub_properties = pNewProperty->GetSubProperties();
-
-        const std::size_t number_of_subproperties = Data["sub_properties"].size();
-
-        // We assign the subproperties now
-        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
-            // Copy of the current parameters
-            Parameters sub_prop = Data["sub_properties"][i_sub_prop];
-
-            // Define subproperties
-            Properties::Pointer p_new_sub_prop = nullptr;
-            if (sub_prop.Has("use_existing_property")) {
-                // We get the adress if any
-                const std::string& r_use_existing_property = sub_prop["use_existing_property"].GetString();
-
-                // We check if already defined
-                bool already_defined = false;
-                if (r_use_existing_property != "") { // NOTE: This means that is not marked as existing
-                    if (r_use_existing_property.size() > 1) {
-                        if (rModelPart.HasProperties(r_use_existing_property, mesh_id)) {
-                            p_new_sub_prop = rModelPart.pGetProperties(r_use_existing_property, mesh_id);
-                            already_defined = true;
-                        }
-                    } else {
-                        const IndexType property_id = std::stoi(r_use_existing_property);
-                        if (rModelPart.RecursivelyHasProperties(property_id)) {
-                            p_new_sub_prop = rModelPart.pGetProperties(property_id);
-                            already_defined = true;
-                        }
+    auto& r_list_sub_properties = rProperty.GetSubProperties();
+
+    const std::size_t number_of_subproperties = SubPropertiesData.size();
+
+    // We assign the subproperties now
+    for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+        // Copy of the current parameters
+        Parameters sub_prop = SubPropertiesData[i_sub_prop];
+
+        // Define subproperties
+        Properties::Pointer p_new_sub_prop = nullptr;
+        if (sub_prop.Has("use_existing_property")) {
+            // We get the adress if any
+            const std::string& r_use_existing_property = sub_prop["use_existing_property"].GetString();
+
+            // TODO why cannot we directly ask the modelpart with the address?
```

Comment:
@loumalouomega why did you split up the following?
Why cannot you directly ask for the property from the modelpart with the address? That would make it much simpler

Reply:
Because this part of the code is previous to that :P